src/hooks/use-imperative-dialog.tsx

```diff
@@ -46,18 +69,31 @@ const DeleteDialog = ({
   )
 }
 
-type ImperativeDialogProps = {
-  heading: string
-  text: string
-  confirmText?: string
-  cancelText?: string
-}
+type ImperativeDialogProps =
```

Comment:
Thought: no need to change anything now, but something to keep in mind if we start adding states and possible combinations to this or to other components following this pattern, we could consider creating a type for each variant of options and combine them to improve readability. i.e. 
```
ImperativeDialogExtraConfirmationProps = 
      | {
          extraConfirmation: true
          entityName: string
        }
      | {
          extraConfirmation: undefined
          entityName: undefined
        }

ImperativeDialogProps =
  | {
      heading: string
      ...
     } & 
     ImperativeDialogExtraConfirmationProps 
```